tests/unit/utils/archive_size.test.ts

```diff
@@ -0,0 +1,29 @@
+import { join, resolve } from 'path'
+
+import { describe, expect, test } from 'vitest'
+
+import { FunctionArchive } from '../../../src/function.js'
+import { addArchiveSize } from '../../../src/utils/archive_size.js'
+import { FIXTURES_DIR } from '../../helpers/main.js'
+
+describe('addArchiveSize', () => {
+  test('adds the archive size of the file at `path` if it is a ZIP archive', async () => {
+    const functionArchive = {
+      path: resolve(join(FIXTURES_DIR, 'archive-size', 'normal.zip')),
```

Comment:
I don't think you need `join` here.

```suggestion
      path: resolve(FIXTURES_DIR, 'archive-size', 'normal.zip'),
```